src/utils/actions.js

```diff
@@ -0,0 +1,254 @@
+// this file provides functions for various action codes
+// it is assumed every function recieves a "context" object
+// Context:
+//  Patient: identifier for the patient whose data triggered the workflow
+//  records: list of records
+//  Encounter: encounter where some data triggered the workflow
+//  PlanDef: *** do we want to keep a "pre-processed" version of the plandef?
+//  Action: action currently in context from the planDef
+//  messageHeader: info about the message
+//  client: set of clients to connect to
+//      source: FHIR client to make additional queries to EHR
+//      dest: FHIR client to send to PHA/TTP
+//      trust services
+//      database
+//  reportingBundle: the entire bundle to be submitted
+//  contentBundle: just the content section of the report bundle
+//  next: place to put arbitrary objects specifically for the next step?
+//  prev: where arbitrary objects from the previous set should go?
+//  flags: various boolean or enum flags about the status of the report. is it encrypted y/n, is it identifiable or de/pseudo/anonymized, etc
+//  cancelToken: indicates the job should be cancelled (from a process external to the job itself -- this tells the job to cease processing)
+//  exitStatus: if set, terminates the job and marks it with the given status (for example, if a given patient doesn't meet reporting criteria, don't keep reporting)
```

Comment:
Based on the actions here can messageHeader, next, and prev be removed? Content IGs may want next and prev